_posts/2019-06-25-building-a-mono-repo.markdown

```diff
@@ -0,0 +1,66 @@
+
+#### 3. Words Mean What I Say They Mean
+Redefine your idea of a build *is* so that you can run through everything on every change. Build every one of your projects for every Pull Request.  How do you get there?
+* Pull long running tests in a separate suite that runs "afterwards". Do you need to run them on every Pull Request? Do they even fail for reasons that aren't flakey? Are they even providing value or are you a test horder?
+* Use incremental builds instead of full/clean. Yes incremental builds can let issues slip through, but how often does that actually happen in your repo? And is it worth the time it costs to perform them. You can always follow up your incrementals with scheduled full builds throughout the day to catch issues that an incremental would miss.
+* Don't use Scala.
+* Configure your test harnesses so that any tests that need more than a single digit number of seconds to complete are marked as failures. Delete any repeat offenders. Anyone who objects due to their importance is responsible for making them faster.
+* Look the other way when that one project that doesn't have any dependencies in the monorepo some how ended up in its own repo.
+* Accept that there is a natural trade off between the speed of your build (and there for your releases) and the amount of things that can happen to give you confidence in the correctness of a pending change.
+
+**_Pros_**: Admittedly, this is not an option for some, dare say many situations. But if you can, teams I've been on that prioritize build/test discipline like this move blazingly fast. Every minute you add between someone hitting merge and their changing reaching production is that much more encouragement to do one Pull Request instead of two or three. This creates larger Pull Requests which give reviewers that much more excuse to drag their feet on getting them done. This incentivizes authors to create even less (but larger) Pull Requests. So your feedback loop keeps turning, productivity it starts burning.
+
+**_Cons_**: Good luck getting here. This is not just a technical decision, but a values decision that your entire team needs to buy into. It only takes one or two folks who aren't comfortable with deleting long running tests before the entire thing is off the rails.
+
+Also, as I previously mentioned, there are lots of types of software for which this just isn't an option. Though if first thought is that you're in this camp I'd ask yourself why? Is the inherent complexity of the software you need to build, the incidental complexity of past design decisions that you'd change if you could, or that you just don't belive it could work?  
```

Comment:
it's not clear if this last line is related to cons, to the 3rd solution, or to the overall blogpost.  Would probably be worth writing a wrap up, and introducing it with formatting. 
belive > believe